src/nspawn/nspawn.c

```diff
@@ -917,23 +916,20 @@ static int parse_argv(int argc, char *argv[]) {
                                 arg_uid_shift = UID_INVALID;
                                 arg_uid_range = UINT32_C(0x10000);
                         } else {
-                                _cleanup_free_ char *buffer = NULL;
                                 const char *range, *shift;
 
                                 /* anything else: User namespacing on, UID range is explicitly configured */
 
                                 range = strchr(optarg, ':');
                                 if (range) {
-                                        buffer = strndup(optarg, range - optarg);
-                                        if (!buffer)
-                                                return log_oom();
-                                        shift = buffer;
+                                        shift = strndupa(optarg, range - optarg);
 
                                         range++;
-                                        if (safe_atou32(range, &arg_uid_range) < 0 || arg_uid_range <= 0) {
-                                                log_error("Failed to parse UID range: %s", range);
-                                                return -EINVAL;
-                                        }
+                                        r = safe_atou32(range, &arg_uid_range);
+                                        if (r < 0)
+                                                return log_error_errno(r, "Failed to parse UID range '%s': %m", range);
+                                        if (arg_uid_range == 0)
+                                                return log_error_errno(EINVAL, "UID range cannot be 0.");
```

Comment:
I know we have this rule, but I'm not convinced it's right. After all
```
     return log_error_errno(EINVAL, "…");
```
is much more concise and thus much clearer then
```
     {
                log_error("…");
                return -EINVAL;
     }
```
I'd say that the practicality of the short form beats the purity of the long one.

Indeed, instead of setting this rule in stone, I'd rather rescind it, and allow judicious use of the shortened form with the hardcoded return errno. The only time where it should *not* be imo allowed, would be when the errno value makes no sense (i..e. when we return ESRCH or ENOMEDIUM because we need to pass some state to the caller and we use a fake errno that is special-cased in the caller).